### PR TITLE
[PF run] Show error msg when pf.run failed

### PR DESCRIPTION
**Pull request by lalala123123**
# Description
warning log
```
[2023-10-12 13:43:07,809][promptflow._sdk.operations._run_submitter][WARNING] - 1 out of 1 runs failed in bulk run. Please check out C:/Users/zhrua/.promptflow/.runs/failed_flow_variant_0_20231012_134245_423704 for more details.
```
Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by elliotzh — DISMISSED** [review]

File: src/promptflow/promptflow/_sdk/operations/_run_submitter.py [hunk 1]
```diff
@@ -295,16 +296,27 @@ def _submit_bulk_run(self, flow: Flow, run: Run, local_storage: LocalStorageOper
         run._dump()  # pylint: disable=protected-access
         try:
             bulk_result = flow_executor.exec_bulk(mapped_inputs, run_id=run_id)
+            # Filter the failed line result
+            failed_line_result = next(
+                filter(lambda result: result.run_info.status == Status.Failed, bulk_result.line_results), None)
+            if failed_line_result:
+                raise UserErrorException(**failed_line_result.run_info.error)
```
Comment (0mza987) [inline src/promptflow/promptflow/_sdk/operations/_run_submitter.py:303]: What will this run status be if error raised here? I don't see the the status set clause in `except` section?

File: src/promptflow/promptflow/_sdk/operations/_run_submitter.py [hunk 2]
```diff
@@ -296,6 +297,16 @@ def _submit_bulk_run(self, flow: Flow, run: Run, local_storage: LocalStorageOper
         run._dump()  # pylint: disable=protected-access
         try:
             bulk_result = flow_executor.exec_bulk(mapped_inputs, run_id=run_id)
+            # Filter the failed line result
+            failed_line_result = \
+                [result for result in bulk_result.line_results if result.run_info.status == Status.Failed]
+            if failed_line_result:
+                # Log warning message when there are failed line run in bulk run.
+                error_log = f"{len(failed_line_result)} run(s) failed in bulk run."
```
Comment (0mza987) [inline src/promptflow/promptflow/_sdk/operations/_run_submitter.py:305]: "x out of y runs failed in bulk run."
In which x means failed runs, y means total runs.

Comment (lalala123123) [reply src/promptflow/promptflow/_sdk/operations/_run_submitter.py]: update warning log

File: src/promptflow/promptflow/_sdk/operations/_run_submitter.py (hunk 1 above)
Comment (lalala123123) [reply src/promptflow/promptflow/_sdk/operations/_run_submitter.py]: revert it, only log warning msg and not change batch run status to failed

**Review by 0mza987 — APPROVED** [review]

**Review by YingChen1996 — APPROVED** [review]

**Pull request merged**